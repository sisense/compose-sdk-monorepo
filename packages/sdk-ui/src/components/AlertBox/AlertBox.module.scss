.alert_box {
  position: absolute;
  display: flex;
  justify-content: center;
  z-index: 1;
  margin-top: 1%;

  .content {
    background-color: #616161;
    padding: 5px;
    border-radius: 3px;
    text-align: center;

    .alert {
      color: white;
      margin: 0 8px;
      line-height: 1;
      font-size: 12px;
    }
  }
}
