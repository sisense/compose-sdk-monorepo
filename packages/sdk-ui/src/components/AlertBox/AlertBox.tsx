import styles from './AlertBox.module.scss';

interface Props {
  alerts: string[];
}

export const AlertBox = ({ alerts = [] }: Props) => {
  return (
    <div className={styles.alert_box}>
      <div className={styles.content}>
        {
          // prevents alert duplication
          [...new Set(alerts)].map((alert, i) => (
            <div className={styles.alert} key={`alert-${i}`}>
              {alert}
            </div>
          ))
        }
      </div>
    </div>
  );
};

export default AlertBox;
