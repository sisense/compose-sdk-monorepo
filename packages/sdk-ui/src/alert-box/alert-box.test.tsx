import { render } from '@testing-library/react';
import AlertBox from './alert-box';

describe('AlertBox component', () => {
  it('renders alerts correctly', () => {
    const alerts = ['Alert 1', 'Alert 2'];
    const { container } = render(<AlertBox alerts={alerts} />);
    expect(container).toMatchSnapshot();
  });

  it('renders no alerts when alerts array is empty', () => {
    const { container } = render(<AlertBox alerts={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('renders no alerts when alerts property is not provided', () => {
    const { container } = render(<AlertBox />);
    expect(container).toMatchSnapshot();
  });
});
