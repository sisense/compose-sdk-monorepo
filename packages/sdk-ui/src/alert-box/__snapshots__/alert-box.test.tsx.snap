// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertBox component > renders alerts correctly 1`] = `
<div>
  <div
    class="_alert_box_619f3c"
  >
    <div
      class="_content_619f3c"
    >
      <div
        class="_alert_619f3c"
      >
        Alert 1
      </div>
      <div
        class="_alert_619f3c"
      >
        Alert 2
      </div>
    </div>
  </div>
</div>
`;

exports[`AlertBox component > renders no alerts when alerts array is empty 1`] = `
<div>
  <div
    class="_alert_box_619f3c"
  >
    <div
      class="_content_619f3c"
    />
  </div>
</div>
`;

exports[`AlertBox component > renders no alerts when alerts property is not provided 1`] = `
<div>
  <div
    class="_alert_box_619f3c"
  >
    <div
      class="_content_619f3c"
    />
  </div>
</div>
`;
